src/components/ShoppingList/ShoppingList.js

```diff
@@ -62,6 +62,90 @@ function ShoppingList({ listId }) {
+      nextPurchaseDate = DateTime.fromSeconds(
+        item.lastPurchaseDate.seconds,
+      ).plus({ days: item.purchaseInterval });
+    } else if (item.createdAt?.seconds) {
+      // if there's no purchase history, estimate it will be bought [purchaseInterval] days from when item was created
+      // (user provides this info at item creation)
+      nextPurchaseDate = DateTime.fromSeconds(item.createdAt.seconds).plus({
+        days: item.purchaseInterval,
+      });
+    } else return null;
+
+    const currentDate = DateTime.fromSeconds(Math.floor(Date.now() / 1000));
+    const daysRemaining = nextPurchaseDate.diff(currentDate, ['days']);
+    return Math.round(daysRemaining.as('days'));
+  };
+
+  const isItemInactive = (item) => {
+    const currentDate = DateTime.fromSeconds(Math.floor(Date.now() / 1000));
+
+    if (item.lastPurchaseDate?.seconds) {
+      // if an item has been purchased before, compare the time elapsed since lastPurchaseDate to the purchaseInterval
+      return (
+        currentDate
+          .diff(DateTime.fromSeconds(item.lastPurchaseDate.seconds), ['days'])
+          .as('days') >=
+        2 * item.purchaseInterval
+      );
+    } else if (item.createdAt?.seconds) {
+      // if an item has never been purchased, do the same calculating using createdAt date instead of lastPurchaseDate
+      return (
+        currentDate
+          .diff(DateTime.fromSeconds(item.createdAt.seconds), ['days'])
+          .as('days') >=
+        2 * item.purchaseInterval
+      );
+    } else return null;
+  };
+
+  const sortListItems = (itemA, itemB) => {
+    // if items being compared are both inactive, or neither is inactive, sort by days until next purchase
+    if (
+      (itemA.status === 'inactive' && itemB.status === 'inactive') ||
+      (itemA.status !== 'inactive' && itemB.status !== 'inactive')
+    ) {
+      if (itemA.daysToPurchase < itemB.daysToPurchase) return -1;
+      if (itemA.daysToPurchase > itemB.daysToPurchase) return 1;
+
+      // if we've made it this far, days to purchase is the same for both items so alphabetize
+      return itemA.itemName.localeCompare(itemB.itemName, 'en', {
+        sensitivity: 'base',
+        ignorePunctuation: true,
+      });
+    } else {
+      // if one item is inactive and the other is not, bump down the inactive item
+      if (itemA.status === 'inactive') return 1;
+      if (itemB.status === 'inactive') return -1;
+    }
+  };
+
+  const getItemStatus = (item) => {
+    if (isItemInactive(item)) return 'inactive';
+    if (item.daysToPurchase < 7) return 'soon';
+    if (item.daysToPurchase >= 7 && item.daysToPurchase <= 30)
+      return 'kind-of-soon';
+    if (item.daysToPurchase > 30) return 'not-soon';
+  };
+
+  // format the data from firestore into a sorted, filtered array of items
+  // add new properties: item.status and item.daysToPurchase
+  const itemsToDisplay = listItems?.docs
```

Comment:
1. I was torn on whether it should be at the top of the component or the bottom so I wouldn't mind moving it, but Nick felt it made sense for it to go right before the JSX. @nick-zanetti what do you think after having another perspective?

2. This is also in the AC, and I think it might help to think of it like this: Let's say you have apples on your list with a purchaseInterval of 8 days and bananas with a purchaseInterval of 15 days. Which should go higher up on the list? You might think apples first, then bananas, since the lower purchaseInterval for apples means you purchase them more often. BUT, what if you just bought apples yesterday so they aren't due for at least a week, and meanwhile you haven't bought bananas for almost 3 weeks so based on your history you're well overdue to buy more? From that perspective, it makes sense to use a combination of the lastPurchaseDate and the purchaseInterval to figure out the next date you're expected to buy a given item. "daysToPurchase" is the number of days from now until that next purchase date, and it is always being recalculated since it changes as time passes.

Reply:
1. I think there are arguments to be made for putting it in either the top of the file or right before the render. Since we are rendering based on `itemsToDispaly`, I thought it made to put it just before that JSX, but I see what you're saying about code readability. Let's move it up top and then put the functions in the order they appear in it, I think that gives the file a nice structure for the future. 

2. So I actually made the mistake of thinking we were sorting by `purchaseInterval` when I first brainstormed this issue. Sorting by the `daysToPurchase` is actually the core functionality of the app as I see it, because it combines the user estimated time interval with actual time that has passed since the purchase, allowing for a "smart" estimate of what should be purchased next 🤯 .

Reply:
Looking at the code, I think it's a good idea to move it to the top. I also think sorting by daysToPurchase is a good idea.